Source/Core/Core/AchievementManager.cpp

```diff
@@ -129,15 +129,82 @@ void AchievementManager::LoadGameByFilenameAsync(const std::string& iso_path,
 
     const auto fetch_game_data_response = FetchGameData();
     m_is_game_loaded = fetch_game_data_response == ResponseType::SUCCESS;
+
+    // Claim the lock, then queue the fetch unlock data calls, then initialize the unlock map in
+    // ActivateDeactiveAchievements. This allows the calls to process while initializing the
+    // unlock map but then forces them to wait until it's initialized before making modifications to
+    // it.
+    m_lock.lock();
+    LoadUnlockData([](ResponseType r_type) {});
+    ActivateDeactivateAchievements();
+    m_lock.unlock();
+    ActivateDeactivateLeaderboards();
+    ActivateDeactivateRichPresence();
+
     callback(fetch_game_data_response);
   });
 }
 
+void AchievementManager::LoadUnlockData(const ResponseCallback& callback)
+{
+  m_queue.EmplaceItem([this, callback] {
+    const auto hardcore_unlock_response = FetchUnlockData(true);
+    if (hardcore_unlock_response != ResponseType::SUCCESS)
+    {
+      callback(hardcore_unlock_response);
+      return;
+    }
+
+    callback(FetchUnlockData(false));
+  });
+}
+
+void AchievementManager::ActivateDeactivateAchievements()
+{
+  for (unsigned ix = 0; ix < m_game_data.num_achievements; ix++)
+  {
+    if (m_unlock_map.count(m_game_data.achievements[ix].id) == 0)
```

Comment:
Consider structuring this entire block scope like:

```cpp
const auto iter = m_unlock_map.insert(ix, UnlockStatus{.game_data_index = ix});
ActivateDeactivateAchievement(iter.first->second.id);
```

`.insert()` only actually performs an insert if there's no element in the map with that key, otherwise it returns the already existing element,